src/litdata/streaming/dataset.py

```diff
@@ -62,15 +62,24 @@ def __init__(
             seed: Random seed for shuffling.
             serializers: The serializers used to serialize and deserialize the chunks.
             max_cache_size: The maximum cache size used by the StreamingDataset.
+            subsample: Float representing fraction of the dataset to be randomly sampled (e.g., 0.1 => 10% of dataset).
 
         """
         super().__init__()
         if not isinstance(shuffle, bool):
             raise ValueError(f"Shuffle should be a boolean. Found {shuffle}")
 
+        if not isinstance(subsample, float) or subsample < 0 or subsample > 1:
+            raise ValueError("subsample must be a float with value between 0 & 1.")
+
         input_dir = _resolve_dir(input_dir)
 
         self.input_dir = input_dir
+        self.chunks: List[Dict[str, Any]] = []
+        self.region_of_interest: List[Tuple[int, int]] = []
+        self.subsampled_files, self.region_of_interest = subsample_streaming_dataset(
```

Comment:
Sorry, I forgot to modify:
```diff
- self.chunks: List[Dict[str, Any]] = []
+ self.subsampled_files: List[str] = []
```

All the relevant subsampling code is in `src/litdata/utilities/dataset_utilities.py` file, and calling function `subsample_streaming_dataset` function.

I believe this already is cleaner, and makes sense. Having it in config file, is weird to me, since, what are the subsampled_files and their roi of a dataset is directly available as it's own property. 

This approach makes it possible to **split a subsampled dataset** or **subsample a splitted dataset** or whatever permutation you wish to do.

---

Now, coming to doing it only when subsample argument is provided, makes complete sense. As we are doing **unnecessary `slightly compute intensive` calculation** of getting roi and chunk_filenames (which are obviously going to be all the files).

But, if we don't do it, config file need to have two methods for loading chunks. And, also, for subsampling dataset we will need to write additional tests to test its functionality. In the current method, all passing tests relevant to dataset justify the subsampling dataset code works.

The one alternative I think is, if subsample argument is 1 (default), don't call the `subsample_streaming_dataset` and just iterate chunks and store all the files and generate `roi`. Everything else remains unchanged.

This will make code faster in the default case saving from some computation.

----

Let me know what you think about my points. Or, a better alternative.